Misc. Changes to Master Branch

crystalsolenoid opened the pull request.

rlhatton commented on ProgramFiles/Utilities/curvature_mode_toolbox/fatbackbone_from_curvature_bases.m:
```diff
@@ -35,10 +45,43 @@
             
+            % Calculate tolerance for find() (this will vary depending on
+                                            % system evaluation density)
+            tol=abs(s.grid.eval{1}(1,1)-s.grid.eval{1}(2,1))*0.5;
```
I believe that what you're doing here is finding a closest grid point to the requested shape, then returning the transformation values at that grid point. Instead, I think you probably want to be using a smooth interpolating function to find the value of the transformation at the requested shape. This would mean you could replace the code from here through line 74 with:

r_cell = mat2cell(r);
x = interpn(s.grid.eval{:}, s.B_optimized.eval.Beta{1},r_cell{:}); % find the value of Bx for the given shape variables
y = interpn(s.grid.eval{:}, s.B_optimized.eval.Beta{2},r_cell{:}); % find the value of By for the given shape variables
theta = interpn(s.grid.eval{:}, s.B_optimized.eval.Beta{3},r_cell{:});% find the value of Btheta for the given shape variables

which also scales cleanly to systems with more than two shape variables.

crystalsolenoid replied:
Done, thanks!

rlhatton commented on ProgramFiles/Utilities/curvature_mode_toolbox/fatbackbone_from_curvature_bases.m:
```diff
@@ -3,14 +3,24 @@
         orientation = 'midpoint-tangent';
+    elseif or(isequal(orientation,'com-mean'), isequal(orientation,'midpoint-tangent'))
+        % do nothing; orientation is a valid option
+    else
```
I don't quite follow your design intent with the logic here -- I would think that pulling from the sys file would be an explicit option, and not coded as a default catching term.

crystalsolenoid replied:
I was trying to avoid changing the how the existing function parameters of illustrate_backbone_shapespace were set up. I used the existing orientation parameter with catching like this because a named orientation like midpoint-tangent can't be used at the same time as the calculated one.

I'm still not sure how function inputs are supposed to work (gracefully) in matlab. Would something like [an input parser](https://www.mathworks.com/help/matlab/ref/inputparser.html) be a good option here?

rlhatton replied:
The ~exist('orientation','var') condition and following code is acting to set the default value for the "orientation argument" if it is not set by the caller (because it is valid matlab syntax to call a function with trailing arguments not specified), and so shouldn't need to be extended from my original code.

I see why you've got the from-sys as a default option -- it allows us to put an arbitrary string in and have it be treated as a system file unless otherwise specified. I would implement this by moving your code from the "else" condition and putting it into an "otherwise" case at the end of the "switch orientation" code block (combining it with code from the "from-sys" case you already have)

rlhatton commented on ProgramFiles/Utilities/curvature_mode_toolbox/fatbackbone_from_general_curvature.m:
```diff
@@ -3,6 +3,18 @@
         orientation = 'midpoint-tangent';
+    elseif or(isequal(orientation,'com-mean'), isequal(orientation,'midpoint-tangent'))
+        % do nothing; orientation is a valid option
+    else
```
Same comment as on backbone from curvature modes

rlhatton commented on ProgramFiles/Utilities/curvature_mode_toolbox/fatbackbone_from_general_curvature.m:
```diff
@@ -33,10 +45,35 @@
+            
+            % Calculate tolerance for find() (this will vary depending on
+                                            % system evaluation density)
+            tol=abs(s.grid.eval{1}(1,1)-s.grid.eval{1}(2,1))*0.5;
```
Same comment as on fatbackbone from curvature modes

crystalsolenoid replied:
Done!

rlhatton commented on ProgramFiles/Utilities/curvature_mode_toolbox/illustrate_backbone.m:
```diff
@@ -0,0 +1,64 @@
+    orientation = 'midpoint-tangent';
+elseif or(isequal(orientation,'com-mean'), isequal(orientation,'midpoint-tangent'))
+    % do nothing; orientation is a valid option
+else
```
Same comment as on fatbackbone structures

rlhatton commented on ProgramFiles/Utilities/curvature_mode_toolbox/illustrate_backbone.m:
```diff
@@ -0,0 +1,64 @@
+function illustrate_backbone(curvdef,paramvalue,orientation,plotnum)
```
Do you know why this is showing up as entirely new and seems to cut off in the middle of the file?

rlhatton commented on ProgramFiles/Utilities/curvature_mode_toolbox/illustrate_backbone_shapespace.m:
```diff
@@ -1,8 +1,22 @@
     orientation = 'midpoint-tangent';
+elseif or(isequal(orientation,'com-mean'), isequal(orientation,'midpoint-tangent'))
+    % do nothing; orientation is a valid option
+else
```
Same comment as above

rlhatton commented on ProgramFiles/Utilities/curvature_mode_toolbox/illustrate_backbone_shapespace.m:
```diff
@@ -14,18 +28,29 @@ function illustrate_backbone_shapespace(curvdef,paramvalues,orientation)
-fh = figure(171);
-
+fh = figure(plotnum);
+set(gcf,'name',orientation);
```
this should be

set(fh,'name',orientation)

since we know the handle for the figure already, and coding it this way avoids a race condition with any other process deciding that it's going to bring a different figure to the top

crystalsolenoid replied:
Done!

rlhatton commented on ProgramFiles/Utilities/curvature_mode_toolbox/illustrate_backbone_shapespace.m:
```diff
@@ -14,18 +28,29 @@ function illustrate_backbone_shapespace(curvdef,paramvalues,orientation)
 axh = axes('Parent',fh);
 axis equal
-hold on
+hold on 
```
Would probably be good to have these explicitly target the axis:

axis(axh,'equal')
hold(axh,'on')

etc. for xticks, yticks, box

crystalsolenoid replied:
Done! I used the newer ``axh.YTick = ...`` approach for xticks, yticks, and box. I think it's more readable, but I can change it to a ``set(axh, 'YTick', ...)`` if you want.

crystalsolenoid commented on ProgramFiles/Utilities/curvature_mode_toolbox/fatbackbone_from_curvature_bases.m:
```diff
@@ -3,14 +3,24 @@
+        % Get the user folder path
+        load('sysplotter_config.mat','inputpath');
+        % The hypothetical file path, supposing orientation is a system name
+        calcfilePath = strcat(inputpath,'\sysplotter_data\sysf_',orientation,'_calc.mat');
```
Done!

crystalsolenoid commented:
Okay, I think I've fixed everything!

rlhatton commented:
I think the same structure appears in several other files, but I don't see them as being updated -- could you check on this?

Thanks!

> On Aug 22, 2018, at 4:58 PM, Quinten Konyn <notifications@github.com> wrote:
> 
> Okay, I think I've fixed everything!
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

crystalsolenoid commented:
You commented about this structure on four different files. I updated three, but I removed the illustrate_backbone function (that I had added earlier). I double checked, and there's no other place in the code I've changed where it appears.
